cpmpy/solvers/z3.py

```diff
@@ -410,7 +410,7 @@ def _z3_expr(self, cpm_con, reify=False):
                 else:
                     lhs = self._z3_expr(lhs)
 
-                if is_boolexpr(rhs):
+                if is_boolexpr(rhs) and not isinstance(rhs,int):
```

Comment:
not sure where `is_boolexpr` is defined, but does it say 'yes' to an int? with fishook, this should be rhs.is_bool()?

Reply:
I think this should be dealt in a different pull request, trying to make is_bool check consistent. Right now, we have many different ways of checking if an expression is bool (is_boolexpr(expr), is_bool(expr), expr.is_bool() etc.).

I can try and make it consistent in a different PR.